libsolidity/formal/Why3Translator.cpp

```diff
@@ -247,7 +247,7 @@ bool Why3Translator::visit(FunctionDefinition const& _function)
 		error(_function, "Only functions inside contracts allowed.");
 	addSourceFromDocStrings(m_currentContract.contract->annotation());
 
-	if (_function.isDeclaredConst())
+	if (_function.isPure() || _function.isView())
```

Comment:
No.  If it  `isView()`, the function should not change the state, so it's reasonable to assert that the new state is equal to the original state.